File: Data Structures & Algorithms/median-of-two-sorted-arrays/submission-0.py
class Solution:
    def findMedianSortedArrays(self, nums1: List[int], nums2: List[int]) -> float:
        nums_combined = []
        n = len(nums1)
        m = len(nums2)
        i = j = 0
        while (i < n and j < m):
            if (nums1[i] <= nums2[j]):
                nums_combined.append(nums1[i])
                i+=1
            else:
                nums_combined.append(nums2[j])
                j+=1
        while (i < n):
            nums_combined.append(nums1[i])
            i += 1
        while (j < m):
            nums_combined.append(nums2[j])
            j += 1
        m_n = n+m
        if (m_n % 2 == 0):
            return (nums_combined[m_n//2] + nums_combined[m_n//2-1])/2
        else:
            return nums_combined[m_n//2]

File: Data Structures & Algorithms/next-greater-element-i/submission-0.cpp
class Solution {
public:
    vector<int> nextGreaterElement(vector<int>& nums1, vector<int>& nums2) {
        int m = nums1.size();
        int n = nums2.size();
        stack<int> stk;
        unordered_map<int, int> tmp_res;
        for (int i=0; i<n; i++) {
            while (!stk.empty() && nums2[i] > nums2[stk.top()]) {
                int idx = stk.top();
                tmp_res[nums2[idx]] = nums2[i];
                stk.pop();
            }
            stk.push(i);
        }
        while (!stk.empty()) {
            tmp_res[nums2[stk.top()]] = -1;
            stk.pop();
        }
        vector<int> res(m, -1);
        for (int i=0; i<m; i++) {
            res[i] = tmp_res[nums1[i]];
        }
        return res;
    }
};